tests/Middleware/MiddlewareTest.php

```diff
@@ -15,50 +15,61 @@
  * @internal
  * @coversNothing
  */
-final class MiddlewareTest extends AbstractTest
+final class MiddlewareTest extends AbstractMiddlewareTest
 {
     /**
-     * Test exception for missing Proxy Ticket in request.
+     * @return mixed
      */
-    public function testNoProxyTicketException()
+    public function proxyTicketCases(): array
     {
-        // Generate response.
-        $content = file_get_contents(self::FIXTURE_DIR . '/create-requests-response.xml');
-        $response = new Response(200, [], $content);
-        $this->httpClient->addResponse($response);
-
-        $clientFactory = $this->createClientFactory();
-        $session = new Session(new MockArraySessionStorage());
-        $middleware = new CasProxyTicketSessionMiddleware($session);
-        $clientFactory->addMiddleware($middleware);
-        $client = $clientFactory->getClient();
-
-        // Try to perform the request but expect an exception.
-        $this->expectExceptionMessage('[epoetry] session has no proxy ticket.');
-        $client->createRequests(new CreateRequests());
+        return $this->getFixture('proxy-ticket-test.yml');
     }
 
     /**
      * Test Proxy Ticket in request.
+     *
+     * @dataProvider proxyTicketCases
      */
-    public function testProxyTicket()
+    public function testProxyTicket(array $input, array $expectations)
     {
+        $input += ['session' => []];
+        $expectations += ['exceptions' => []];
+
         // Generate response.
-        $content = file_get_contents(self::FIXTURE_DIR . '/create-requests-response.xml');
-        $response = new Response(200, [], $content);
+        $response = new Response(200, [], $input['xml']);
         $this->httpClient->addResponse($response);
 
         $clientFactory = $this->createClientFactory();
         $session = new Session(new MockArraySessionStorage());
-        $session->set('cas_pgt', 'DESKTOP_PT-21-9fp9');
+
+        $values = [
+            'session' => $session,
+        ];
+
+        foreach ($input['session'] as $expression) {
+            $this->expressionLanguage->evaluate($expression, $values);
+        }
+
         $middleware = new CasProxyTicketSessionMiddleware($session);
         $clientFactory->addMiddleware($middleware);
         $client = $clientFactory->getClient();
+
+        foreach ($expectations['exceptions'] as $expression) {
+            $this->expectExceptionMessage($expression);
+        }
+
         $client->createRequests(new CreateRequests());
 
-        // Perform request.
-        $request = $client->debugLastSoapRequest()['request'];
+        $values['request'] = $client->debugLastSoapRequest()['request'];
 
-        $this->assertContains('ecas:ProxyTicket: DESKTOP_PT-21-9fp9', $request['headers'], 'Request XML header malformed, missing proxy ticket.');
+        foreach ($expectations['assertions'] as $expression) {
```

Comment:
This should be moved into its own method as it's repeated 3 times in different test classes.

Reply:
Fixed.